src/checkers/inference/util/JaifBuilder.java

```diff
@@ -123,6 +123,9 @@ private String buildAnnotationHeader(Class<? extends Annotation> annotation) {
                 }
                 if (method.getReturnType().isArray()) {
                     result += method.getReturnType().getComponentType().getSimpleName() + "[]";
+                } else if (method.getReturnType().isPrimitive() || method.getReturnType()
+                        .getCanonicalName().contentEquals("java.lang.String")) {
```

Comment:
Why it is special for `java.lang.String`?

Could you add some comments to illustrate that?

Why `JaifBuilder` doesn't have problems with other reference types?

Reply:
Explained by Jeff in person.

Reply:
Adding a TODO to rewrite this method from scratch. We don't account for all possible cases of outputs at this moment (eg annotation-field arrays). A clean rewrite should exercise and be tested for all valid annotation field types.